### PR TITLE
Add the view all tags & categories screen

### PR DESCRIPTION
**Pull request by 0nko**
This PR adds a new View all tags & categories screen.

To test:
1. Go to Stats
2. Scroll down to the Tags & Categories block
3. Tap View more
4. Notice a new screen opens up with more tags

Note: This PR can only be merged after the Comments view all screen is merged and the base branch is updated.

image

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/StatsModule.kt [hunk 1]
```diff
@@ -173,6 +175,27 @@ class StatsModule {
         }
     }
 
+    /**
+     * Provides a singleton TagsAndCategoriesUseCase that represents the Followers View all screen
```
Comment (malinajirka) [inline WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/StatsModule.kt]: Out-dated comment

File: WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/lists/sections/insights/usecases/TagsAndCategoriesUseCase.kt [hunk 2]
```diff
@@ -23,40 +26,45 @@ import org.wordpress.android.ui.stats.refresh.lists.sections.BlockListItem.Link
...
     private val resourceProvider: ResourceProvider,
-    private val analyticsTracker: AnalyticsTrackerWrapper
+    private val analyticsTracker: AnalyticsTrackerWrapper,
+    useCaseMode: UseCaseMode
 ) : StatefulUseCase<TagsModel, TagsAndCategoriesUiState>(
         TAGS_AND_CATEGORIES,
         mainDispatcher,
         TagsAndCategoriesUiState(null)
 ) {
+    private val itemsToLoad = if (useCaseMode == VIEW_ALL) VIEW_ALL_ITEM_COUNT else BLOCK_ITEM_COUNT
```
Comment (malinajirka) [inline WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/lists/sections/insights/usecases/TagsAndCategoriesUseCase.kt]: As per my comment in the #9197 PR -> We might want to replace `when` with `if`. Wdyt?

File: WordPress/src/test/java/org/wordpress/android/ui/stats/refresh/lists/sections/insights/usecases/TagsAndCategoriesUseCaseTest.kt [hunk 3]
```diff
@@ -71,7 +74,7 @@ class TagsAndCategoriesUseCaseTest : BaseUnitTest() {
                 listOf(firstTag, secondTag),
                 categoryViews
         )
-        whenever(insightsStore.fetchTags(site, pageSize, forced)).thenReturn(
+        whenever(insightsStore.fetchTags(site, FetchMode.Top(pageSize), forced)).thenReturn(
```
Comment (malinajirka) [inline WordPress/src/test/java/org/wordpress/android/ui/stats/refresh/lists/sections/insights/usecases/TagsAndCategoriesUseCaseTest.kt]: Nitpick: We might want to consider renaming "pageSize" to "blockItemCount" to keep it consistent with "BLOCK_ITEM_COUNT" constant from the UseCase. Wdyt?

**Review by malinajirka — COMMENTED** [review]
Thanks @0nko! The code looks good.

However, I have a similar issue as in the Comments PR (#9197). I always see maximum of 7 items on the ViewAll screen. However, it works as expected on version 10.9.

**Comment by SylvesterWilmott** [comment]
I've left some feedback on #9197 that is relevant here

**Comment by 0nko** [comment]
@SylvesterWilmott Thanks, we've decided to address the style changes in a separate PR because it affects all the existing View more screens.

@malinajirka Ready for another pass.

File: WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/lists/sections/insights/usecases/TagsAndCategoriesUseCase.kt (hunk 2 above)
Comment (0nko) [reply WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/lists/sections/insights/usecases/TagsAndCategoriesUseCase.kt]: Please see my reply in #9197.

**Review by malinajirka — APPROVED** [review]
Thanks! LGTM

**Pull request merged**